internal/builder/time.go

```diff
@@ -19,22 +20,32 @@ import (
 // TODO: consider replacing this with AvalancheGo block build metering
 const minBuildGap int64 = 25 // ms
 
-var _ Builder = (*Time)(nil)
+type Mempool interface {
+	Len(context.Context) int // items
+}
+
+type GetPreferedTimestampAndBlockGap func(int64) (int64, int64)
```

Comment:
nit: could we add variable names and a comment to make this more easily readable and an error, so that we don't need to use a special case of the latter return value?

Reply:
sure!